clang-22 build issue

Hello,

In the very specific case of clang-22 + gcc-15 on ubuntu 26.04:

```

/__w/score/score/3rdparty/llfio/include/llfio/v2.0/path_view.hpp:1059:19: error: use of variable template '_is_allocator_based' requires template arguments
 1059 |     static_assert(_is_allocator_based<> + _is_deleter_based<> == 1,
      |                   ^
/__w/score/score/3rdparty/llfio/include/llfio/v2.0/path_view.hpp:3224:60: note: in instantiation of template class 'llfio_v2_f386fd6b::path_view_component::rendered_path<llfio_v2_f386fd6b::path_view_component::not_zero_terminated, char, std::default_delete<char[]>>' requested here
 3224 |   path_view_component::not_zero_terminated_rendered_path<> zpath(b);
      |                                                            ^
/__w/score/score/3rdparty/llfio/include/llfio/v2.0/path_view.hpp:1058:53: note: template is declared here
 1058 |     template <class X = void> static constexpr bool _is_allocator_based = std::is_void<deleter_type>::value;
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~                       ^
/__w/score/score/3rdparty/llfio/include/llfio/v2.0/path_view.hpp:1059:19: error: use of variable template '_is_allocator_based' requires template arguments
 1059 |     static_assert(_is_allocator_based<> + _is_deleter_based<> == 1,
      |                   ^
/__w/score/score/3rdparty/llfio/include/llfio/v2.0/detail/impl/posix/path_handle.ipp:35:48: note: in instantiation of template class 'llfio_v2_f386fd6b::path_view_component::rendered_path<llfio_v2_f386fd6b::path_view_component::zero_terminated, char, std::default_delete<char[]>>' requested here
   35 |     path_view::zero_terminated_rendered_path<> zpath(path);
      |                                                ^
/__w/score/score/3rdparty/llfio/include/llfio/v2.0/path_view.hpp:1058:53: note: template is declared here
 1058 |     template <class X = void> static constexpr bool _is_allocator_based = std::is_void<deleter_type>::value;
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~                       ^
/__w/score/score/3rdparty/llfio/include/llfio/v2.0/path_view.hpp:1059:19: error: use of variable template '_is_allocator_based' requires template arguments
 1059 |     static_assert(_is_allocator_based<> + _is_deleter_based<> == 1,
      |                   ^
/usr/lib/gcc/x86_64-linux-gnu/15/../../../../include/c++/15/bits/stl_vector.h:376:35: note: in instantiation of template class 'llfio_v2_f386fd6b::path_view_component::rendered_path<llfio_v2_f386fd6b::path_view_component::zero_terminated, char, std::default_delete<char[]>, 1>' requested here
  376 |                       _M_impl._M_end_of_storage - _M_impl._M_start);
      |                                                 ^

```

It's the only case failing in my ubuntu matrix:

```
          - { deploy: false, distro: ubuntu.noble,  image: noble,  CMAKEFLAGS: "-DSCORE_PCH=1",         CC: gcc-14,              CXX: g++-14,              CFLAGS: "-O0 -g0", LDFLAGS: "-fuse-ld=lld"   , PKGS: "" }
          - { deploy: true,  distro: ubuntu.noble,  image: noble,  CMAKEFLAGS: "-DCMAKE_UNITY_BUILD=1", CC: clang-19,            CXX: clang++-19,          CFLAGS: "-O3 -g0", LDFLAGS: "-fuse-ld=lld-19", PKGS: "" }
          - { deploy: false, distro: ubuntu.plucky, image: plucky, CMAKEFLAGS: "-DSCORE_PCH=1",         CC: gcc,                 CXX: g++,                 CFLAGS: "-O0 -g0", LDFLAGS: ""               , PKGS: "" }
          - { deploy: true,  distro: ubuntu.plucky, image: plucky, CMAKEFLAGS: "-DCMAKE_UNITY_BUILD=1", CC: clang-20,            CXX: clang++-20,          CFLAGS: "-O3 -g0", LDFLAGS: "-fuse-ld=lld-20", PKGS: "" }
          - { deploy: false, distro: ubuntu.2604,   image: 26.04,  CMAKEFLAGS: "-DSCORE_PCH=1",         CC: gcc,                 CXX: g++,                 CFLAGS: "-O0 -g0", LDFLAGS: ""               , PKGS: "libclang-dev llvm-dev" } # GCC comes with build-essential
          - { deploy: true,  distro: ubuntu.2604,   image: 26.04,  CMAKEFLAGS: "-DCMAKE_UNITY_BUILD=1", CC: clang,               CXX: clang++,             CFLAGS: "-O3 -g0", LDFLAGS: "-fuse-ld=lld"   , PKGS: "clang lld libclang-dev llvm-dev" }
          - { deploy: false, distro: ubuntu.2604,   image: 26.04,  CMAKEFLAGS: "-DSCORE_PCH=1",         CC: gcc-16,              CXX: g++-16,              CFLAGS: "-O0 -g0", LDFLAGS: ""               , PKGS: "gcc-16 g++-16 libclang-dev llvm-dev" }
          - { deploy: false, distro: ubuntu.2604,   image: 26.04,  CMAKEFLAGS: "-DCMAKE_UNITY_BUILD=1", CC: clang-22,            CXX: clang++-22,          CFLAGS: "-O0 -g0", LDFLAGS: "-fuse-ld=lld-22", PKGS: "clang-22 lld-22 libclang-22-dev llvm-22-dev" }

```

--- a/include/llfio/v2.0/path_view.hpp
+++ b/include/llfio/v2.0/path_view.hpp
@@ -1056,7 +1056,7 @@ public:
   private:
     template <class X = void> static constexpr bool _is_deleter_based = std::is_void<allocator_type>::value;
     template <class X = void> static constexpr bool _is_allocator_based = std::is_void<deleter_type>::value;
-    static_assert(_is_allocator_based<> + _is_deleter_based<> == 1,
+    static_assert(_is_allocator_based<void> + _is_deleter_based<void> == 1,
                   "AllocatorOrDeleter must be either a callable deleter, or a STL allocator, for value_type");
 
     LLFIO_TEMPLATE(class U, class source_type)

